### PR TITLE
Shoot photos with volume buttons

### PR DESCRIPTION
**Pull request by mdb1**
With this code, users will have the possibility to enable or disable the option to shoot a photo with the volume buttons.

Its true by default

### CODE REVIEW

**Comment by s4cha** [comment]
@mdb1 
That looks like a great addition! Thanks a lot for taking the time to issue the PR, this is really appreciated :)
Reviewing it at the moment 🚀

**Comment by s4cha** [comment]
A few remarks:

- Code looks good overall 👍 
- The shot is not triggered on my side (iPhone X) 
the notification I receive is 
```
NSConcreteNotification 0x282a95860 {name = AVSystemController_SystemVolumeDidChangeNotification; object = <AVSystemController: 0x2827151d0>; userInfo = {
    "AVSystemController_AudioCategoryNotificationParameter" = "Audio/Video";
    "AVSystemController_AudioVolumeChangeReasonNotificationParameter" = ExplicitVolumeChange;
    "AVSystemController_AudioVolumeNotificationParameter" = "0.75";
    "AVSystemController_UserVolumeAboveEUVolumeLimitNotificationParameter" = 0;
}}
```
So the key `"AVSystemController_SystemVolumeDidChangeNotification` is not present.
I've never worked with this api before so can't really be more helpful than that 😄 

Things to look for that come to my mind:
- Taping the volume button when showing the library screen should not shot a pic, and should show the native volume view as I believe the current code disables it altogether.
- Implementing for video as well would be great :) (though don't feel like you have to )

Hope this helps,

**Comment by mdb1** [comment]
@s4cha Its working now on my iPhone X running iOS 12, let me know if it shoots in yours aswell and I will address the other issues

**Comment by s4cha** [comment]
@mdb1 I confirm that it now triggers the shoot, it also triggers a crash when showing either library or video :)

**Comment by s4cha** [comment]
@mdb1 let me know if you need a hand on this :)

File: Source/Pages/Photo/YPCameraVC.swift [hunk 1]
```diff
@@ -128,6 +144,15 @@ public class YPCameraVC: UIViewController, UIGestureRecognizerDelegate, YPPermis
         }
     }
     
+    @objc
+    func volumeChanged(notification: NSNotification) {
+        if let userInfo = notification.userInfo,
+            let volumeChangeType = userInfo[volumeNotificationChangeKey] as? String,
+            volumeChangeType == "ExplicitVolumeChange" {
+            self.shoot()
```
Comment (NikKovIos) [inline Source/Pages/Photo/YPCameraVC.swift:152]: Maybe need to add something to prevent double tap?

File: Source/Pages/Photo/YPCameraVC.swift [hunk 2]
```diff
@@ -38,6 +42,18 @@ public class YPCameraVC: UIViewController, UIGestureRecognizerDelegate, YPPermis
         let tapRecognizer = UITapGestureRecognizer(target: self, action: #selector(self.focusTapped(_:)))
         tapRecognizer.delegate = self
         v.previewViewContainer.addGestureRecognizer(tapRecognizer)
+        
+        // Shoot photo with volume buttons
+        if YPConfig.shootsWithVolumeButtons {
+            let volumeView = MPVolumeView(frame: CGRect(x: -20.0, y: -20.0, width: 0.0, height: 0.0))
+            self.view.addSubview(volumeView)
+            NotificationCenter.default.addObserver(
```
Comment (NikKovIos) [inline Source/Pages/Photo/YPCameraVC.swift:50]: Don't see where u're removing the observer.

**Pull request closed without merging**